flight/Modules/VtolPathFollower/vtol_follower_control.c

```diff
@@ -87,107 +157,156 @@ int32_t vtol_follower_control_path(const float dT, const PathDesiredData *pathDe
+	// XXX parameterize
+	float error_speed = vtol_deadband(progress->error, 1.25, 0.1) * 
 	    guidanceSettings.HorizontalPosPI[VTOLPATHFOLLOWERSETTINGS_HORIZONTALPOSPI_KP];
-	float correction_velocity[2] = {progress->correction_direction[0] * error_speed, 
-	    progress->correction_direction[1] * error_speed};
-	
-	float total_vel = sqrtf( powf(correction_velocity[0], 2) + powf(correction_velocity[1], 2) );
-	float scale = 1;
-	if (total_vel > guidanceSettings.HorizontalVelMax)
-		scale = guidanceSettings.HorizontalVelMax / total_vel;
-
-	// Currently not apply a PID loop for horizontal corrections
-	velocityDesired.North += progress->correction_direction[0] * error_speed * scale;
-	velocityDesired.East += progress->correction_direction[1] * error_speed * scale;
+
+	/* Sum the desired path movement vector with the correction vector */
+	commands_ned[0] = progress->path_direction[0] * groundspeed +
+	    progress->correction_direction[0] * error_speed;
 	
-	// Interpolate desired velocity along the path
-	float altitudeSetpoint = pathDesired->Start[2] + (pathDesired->End[2] - pathDesired->Start[2]) *
-	    bound_min_max(progress->fractional_progress, 0, 1);
+	commands_ned[1] = progress->path_direction[1] * groundspeed +
+	    progress->correction_direction[1] * error_speed;
 
-	float downError = altitudeSetpoint - positionActual.Down;
-	velocityDesired.Down = pid_apply_antiwindup(&vtol_pids[DOWN_POSITION], downError,
-		-guidanceSettings.VerticalVelMax, guidanceSettings.VerticalVelMax, dT);
+	/* Limit the total velocity based on the configured value. */
+	vtol_limit_velocity(commands_ned, guidanceSettings.HorizontalVelMax);
+
+	VelocityDesiredData velocityDesired;
+
+	velocityDesired.North = commands_ned[0];
+	velocityDesired.East = commands_ned[1];
+	velocityDesired.Down = commands_ned[2];
 
 	VelocityDesiredSet(&velocityDesired);
 
 	return 0;
 }
 
-/**
- * Control algorithm to stay or approach at a fixed location.
- * @param[in] dT the time since last evaluation
- * @param[in] ned The position to attempt to hold
- * This method does not attempt any particular path, simply a straight line
- * approach. The calculated velocity to attempt is stored in @ref 
- * VelocityDesired. 
- *
- * Takes in @ref PositionActual and compares it to @ref PositionDesired 
- * and computes @ref VelocityDesired
- */
-int32_t vtol_follower_control_endpoint(const float dT, const float *hold_pos_ned)
-{
+int32_t vtol_follower_control_simple(const float dT, const float *hold_pos_ned,
+	int landing) {
 	PositionActualData positionActual;
 	VelocityDesiredData velocityDesired;
 	
 	PositionActualGet(&positionActual);
-	VelocityDesiredGet(&velocityDesired);
-	
-	float northError;
-	float eastError;
-	float downError;
-	float northCommand;
-	float eastCommand;
 	
-	const float cur_pos_ned[3] = { positionActual.North, positionActual.East, positionActual.Down };
+	VelocityActualData velocityActual;
+
+	VelocityActualGet(&velocityActual);
+
+	// XXX parameterize
```

Comment:
yup definitely parameterize. probably with default of 0 until we get practical testing experience.